torch/distributed/fsdp/_init_utils.py

```diff
@@ -395,9 +395,7 @@ def _init_device_handle(
                         f"FSDP does not support modules with different device types "
                         f"but got params on {determined_device.type} and {param.device.type}"
                     )
-        determined_device = determined_device or torch.device(
-            "cuda", torch.cuda.current_device()
-        )
+        determined_device = determined_device or torch._C._get_accelerator(check=True)
```

Comment:
Not sure what the error will be here exactly if it is a cpu-only build. We might want to raise an FSDP-specific error message to make it nicer to the end user instead?

Reply:
@albanD I have tried the cpu-only build. For the origin code(use cuda default), it just raise `AssertionError: Torch not compiled with CUDA enabled`, I have added a FSDP-specific error message here.